### ISSUE TITLE
Type hint inspection is not considering PHPDoc in implemented interfaces

### ISSUE THREAD

**Issue by marcospassos** [phpstan/phpstan-strict-rules#49]
The strict rule reports `Method HelloWorld::unserialize() has parameter $serialized with no typehint` for the following snippet:

```php
<?php declare(strict_types = 1);

class HelloWorld implements \Serializable
{
	public function serialize() : string
    {
        return \serialize('foo');
    }

    /**
     * {@inheritdoc}
     */
    public function unserialize($serialized) : void
    {

    }
}
```

**Comment by CZechBoY** [comment]
Typehint or PHPDoc, the message is confusing...

https://phpstan.org/r/8c0f7611-dec6-4c02-b55a-ca8d6640fd86 this should be fine, but phpstan complains about contravariation...

@ondrejmirtes Do you know where is the problem?

**Comment by ondrejmirtes** [comment]
mixed -> string is not contravariant, that's why it's a problem.

**Comment by ondrejmirtes** [comment]
mixed -> string is not contravariant, that's why it's a problem.

Please send a PR that updates `Serializable::unserialize` in functionMap.php. Thanks.

**Comment by ondrejmirtes** [comment]
Thanks. Please verify that it now works for your example.
